### ISSUE TITLE
Loosen upper bounds on distributed-process-X packages

### ISSUE THREAD

**Issue by tdietert** [haskell-distributed/distributed-process-supervisor#10]
I'm trying to build this project with the latest version of `distributed-process` (0.7.3) and currently need to increase the upper-bounds of several packages that this package depends on, and the bounds of some packages _they_ depend on.

The dependency bounds I'm interested in changing are:
```
- distributed-process <0.7 to  <=0.7.3
- ansi-terminal <0.7 to <=0.8.0.4
- time <1.7 to <=1.8.0.2
- network-transport ==0.4 to ==0.5.2
- network-transport-tcp <0.6 to <=0.6
```
The only two of these that seem to affect extra-deps (`distributed-process-X` packages) directly are `time` and `distributed-process`, which obviously also have some dependency constraints that would need reconciling.

Would you be open to me submitting sweeping PRs that made these bounds changes across all the relevant `distribute-process-X` packages:
```
- distributed-process-extras-0.3.2 
- distributed-process-async-0.2.4  
- distributed-process-systest-0.1.1 
```
as well as _this_ one, `distributed-process-supervisor`?

I'm really interested in using this package, and will for the mean time use `allow-newer: true` with stack so that I can build this library in a codebase that uses `distributed-process-0.7.3`, but would love to get these upper bounds fixed upstream and depend on these repos directly, without the use of `allow-newer`. Thanks!

**Comment by qnikst** [comment]
You should not ask about permission to send PR! PR for dependency loosening  are welcome. And will be merged and new version and(or) revision  will be released.

**Comment by tdietert** [comment]
Great, PRs will be incoming within the next week :+1:

**Comment by portnov** [comment]
Hi all.
@tdietert how things are going? Do you need help with this PR ? :)
Similar problem exists with client-server package.

**Comment by tdietert** [comment]
I didn't end up having the time as the abstraction I'm building on top of `distributed-process-supervisor` ended up taking up all my time (and the project built just fine with `allow-newer: true` in the stack.yaml file as a temporary fix. 

My coworker @Acentelles may submit these PRs next week as I am out of the office for the next two weeks.

**Comment by tdietert** [comment]
resolved with PR #11 
@portnov you should be able to build both libs now with `distributed-process-0.7.4` :heart:
